### PR TITLE
applyFilter() made protected

### PR DESCRIPTION
**Pull request by ArkaprabhaChakraborty**
Fixes Issue: 
Runtime error :: PCL bilateral filter #4676

### CODE REVIEW

**Comment by larshg** [comment]
Lets move the protected section down, so its above the private section.
Then we have clearer separation.

**Comment by ArkaprabhaChakraborty** [comment]
Okay I'll move that down

**Comment by mvieth** [comment]
You are moving the wrong documentation. The documentation of a function is always directly _before_ the function

**Comment by ArkaprabhaChakraborty** [comment]
> You are moving the wrong documentation. The documentation of a function is always directly _before_ the function
sorry I forgot the documentation part. I will make the changes.

**Comment by mvieth** [comment]
You are still moving documentation of another function. The documentation of `computePointWeight` must stay where it is!

**Comment by ArkaprabhaChakraborty** [comment]
> You are still moving documentation of another function. The documentation of `computePointWeight` must stay where it is!

So sorry I didn't see the computePointWeight documentation was also removed.

File: filters/include/pcl/filters/bilateral.h [hunk 1]
```diff
@@ -120,8 +112,20 @@ namespace pcl
         */
       inline void
       setSearchMethod (const KdTreePtr &tree)
-      { tree_ = tree; }
+      { tree_ = tree; }
```
Comment (mvieth) [inline filters/include/pcl/filters/bilateral.h:115]: You added spaces at the end of this line. Please undo that.
You also added quite many empty lines before and after the moved function. One empty line for separation is enough. Sorry for being so pedantic about this, but the code style should stay consistent.

Comment (ArkaprabhaChakraborty) [reply filters/include/pcl/filters/bilateral.h:115]: yep got it

**Review by mvieth — APPROVED** [review]
Now it looks good. Thank you!

**Review by larshg — APPROVED** [review]

**Comment by ArkaprabhaChakraborty** [comment]
Btw I have an odd request. Can this PR be merged as a "hactoberfest" PR. :) 
I mean if it is merged in october or so :) Lile please :) if possible :)

**Comment by mvieth** [comment]
> Btw I have an odd request. Can this PR be merged as a "hactoberfest" PR. :)
> I mean if it is merged in october or so :) Lile please :) if possible :)

We can't promise that: When merging a PR, we have to consider whether it breaks ABI or API. This PR breaks API and (I believe, not 100% sure) also ABI, so we will merge it just before releasing PCL 1.13.0. The next PCL release will be 1.12.1, which should not break API or ABI compared to PCL 1.12.0.
BTW: I am not super familiar with the hacktoberfest rules, but don't the PRs have to be _opened_ in October to be counted?

**Comment by ArkaprabhaChakraborty** [comment]
> > Btw I have an odd request. Can this PR be merged as a "hactoberfest" PR. :)
> > I mean if it is merged in october or so :) Lile please :) if possible :)
> 
> We can't promise that: When merging a PR, we have to consider whether it breaks ABI or API. This PR breaks API and (I believe, not 100% sure) also ABI, so we will merge it just before releasing PCL 1.13.0. The next PCL release will be 1.12.1, which should not break API or ABI compared to PCL 1.12.0.
Okay got it. 

> BTW: I am not super familiar with the hacktoberfest rules, but don't the PRs have to be _opened_ in October to be counted?

I guess no it just needs to merged with "hactoberfest-accepted" tag, but if that's not the case then no problem :). If there are any other problems that I can solve for pcl 1.12.1 or 1.12.2 other than the ones that Im owrking on, I would be happy to contribute :)

**Comment by mvieth** [comment]
> If there are any other problems that I can solve for pcl 1.12.1 or 1.12.2 other than the ones that Im working on, I would be happy to contribute :)

Issue #2460 looks like it could probably be merged for PCL 1.12.1 (no guarantee on that though, I also can't guarantee how fast we will be with reviewing)

**Pull request merged**